src/epochs.lua

```diff
@@ -344,18 +351,20 @@ function epochs.computePrescribedObserversForEpoch(epochIndex, hashchain)
 	return prescribedObserversLookup, weightedGateways
 end
 
---- Gets the epoch timestamps for an epoch index
+--- Gets the epoch timestamps for an epoch index. Epochs are 0-indexed.
 --- @param epochIndex number The epoch index
---- @return number, number, number # 	The epoch start timestamp, epoch end timestamp, and epoch distribution timestamp
+--- @return number, number # 	The epoch start timestamp, epoch end timestamp
 function epochs.getEpochTimestampsForIndex(epochIndex)
+	if epochIndex < 0 then
+		return 0, 0
+	end
 	local epochStartTimestamp = epochs.getSettings().epochZeroStartTimestamp
```

Comment:
All of these calls to getSettings make a deep copy of the settings. Consider introducing an unsafe variant.